File: HFSGuardaDiretorioVS2022/HFSGuardaDiretorio_CSharp/objetosdao/AbaDAO.cs
/*
 * Created by SharpDevelop.
 * User: henrique.souza
 * Date: 03/07/2015
 * Time: 17:27
 * 
 * To change this template use Tools | Options | Coding | Edit Standard Headers.
 */
using System;
using System.Data;
using System.Data.SQLite;
using System.Collections.Generic;
using HFSGuardaDiretorio.comum;
using HFSGuardaDiretorio.objetos;

namespace HFSGuardaDiretorio.objetosdao
{
	/// <summary>
	/// Description of AbaDAO.
	/// </summary>
	public sealed class AbaDAO
	{
		private static AbaDAO instancia = new AbaDAO();
		
	    private SQLiteConnection conexao;

	    private SQLiteCommand cmd;
	    
	    private SQLiteDataReader dr = null;
		
		public static AbaDAO Instancia {
			get {
				return instancia;
			}
		}
		
		private AbaDAO()
		{
		}
		
	    private void fecharSqlPreparado() {
	        if (dr != null) {
	            dr.Close();
	        }
	    }
	
	    private Aba novoObjeto(SQLiteDataReader dr) {
	        Aba ret = new Aba();
	        ret.Codigo = dr.GetInt32(0);
	        ret.Nome = dr.GetString(1);
	        return ret;
	    }
	
	    public int consultarTotal() {
	        int total = 0;
	        conexao = Rotinas.getConexao();
			cmd = new SQLiteCommand("select count(*) from Abas", conexao);	
			dr = cmd.ExecuteReader();
	
	        if (dr.Read()) {
	            total = dr.GetInt32(0);
	        }
	        this.fecharSqlPreparado();
	
	        return total;
	    }
	
	    public List<Aba> consultarTudo(IProgressoLog progressoLog) {
	        Aba obj;
	        List<Aba> lista = new List<Aba>();
	        Progresso pb;
	        int i = 0;
	        int total = this.consultarTotal();
	
	        conexao = Rotinas.getConexao();
	        cmd = new SQLiteCommand("select cod, nome from Abas", conexao);	
			dr = cmd.ExecuteReader();
	        	
	        if (total > 0) {
	            pb = new Progresso();
	            pb.Minimo = 0;
	            pb.Maximo = total - 1;
	            pb.Posicao = 0;
	            pb.Passo = 1;
	
	            while (dr.Read()) {
	                obj = novoObjeto(dr);
	                obj.Ordem = i + 1;
	
	                lista.Add(obj);
	
	                pb.Posicao = i;
	
	                if (progressoLog != null) {
	                    progressoLog.ProgressoLog(pb);
	                }
	
	                i++;
	            }
	        }
	
	        this.fecharSqlPreparado();
	
	        return lista;
	    }
	
	    public int incluir(Aba obj) {
	        int ret;
	        conexao = Rotinas.getConexao();
	        cmd = new SQLiteCommand("insert into Abas(cod, nome) values(@1,@2)", conexao);	        
	        cmd.Parameters.AddWithValue("@1", obj.Codigo);
	        cmd.Parameters.AddWithValue("@2", obj.Nome);
	        cmd.Prepare();
	        ret = cmd.ExecuteNonQuery();
	        return ret;
	    }
	
	    public int alterar(Aba obj) {
	        int ret;
	        conexao = Rotinas.getConexao();
	        cmd = new SQLiteCommand("update Abas set nome=@1 where cod=@2", conexao);
	        cmd.Parameters.AddWithValue("@1", obj.Nome);
	        cmd.Parameters.AddWithValue("@2", obj.Codigo);
	        cmd.Prepare();
	        ret = cmd.ExecuteNonQuery();
	        return ret;
	    }
	
	    public int excluir(int codigo) {
	        int ret;
	        conexao = Rotinas.getConexao();
			
	        cmd = new SQLiteCommand("delete from Diretorios where aba=@1", conexao);
	        cmd.Parameters.AddWithValue("@1", codigo);
	        cmd.Prepare();
	        cmd.ExecuteNonQuery();
	
	        cmd = new SQLiteCommand("delete from Abas where cod=@1", conexao);
	        cmd.Parameters.AddWithValue("@1", codigo);
	        cmd.Prepare();
	        ret = cmd.ExecuteNonQuery();
	        
	        return ret;
	    }
	
	    public int criarTabela() {
	        int ret;
	        conexao = Rotinas.getConexao();
	        cmd = new SQLiteCommand("create table IF NOT EXISTS Abas("
	                + "cod integer not null, nome varchar(10) not null, "
	                + "primary key (cod))", conexao);
	        ret = cmd.ExecuteNonQuery();
	        return ret;
	    }
		
	}
}

File: HFSGuardaDiretorioVS2022/HFSGuardaDiretorio_VBNet/objetosdao/AbaDAO.vb
'
' * Created by SharpDevelop.
' * User: henrique.souza
' * Date: 03/07/2015
' * Time: 17:27
' * 
' * To change this template use Tools | Options | Coding | Edit Standard Headers.
' 

Imports System.Data
Imports System.Data.SQLite
Imports System.Collections.Generic
Imports HFSGuardaDiretorio.comum
Imports HFSGuardaDiretorio.objetos

Namespace objetosdao
	''' <summary>
	''' Description of AbaDAO.
	''' </summary>
	Public NotInheritable Class AbaDAO
		Private Shared m_instancia As New AbaDAO()

		Private conexao As SQLiteConnection

		Private cmd As SQLiteCommand

		Private dr As SQLiteDataReader = Nothing

		Public Shared ReadOnly Property Instancia() As AbaDAO
			Get
				Return m_instancia
			End Get
		End Property

		Private Sub New()
		End Sub

		Private Sub fecharSqlPreparado()
			If dr IsNot Nothing Then
				dr.Close()
			End If
		End Sub

		Private Function novoObjeto(dr As SQLiteDataReader) As Aba
			Dim ret As New Aba()
			ret.Codigo = dr.GetInt32(0)
			ret.Nome = dr.GetString(1)
			Return ret
		End Function

		Public Function consultarTotal() As Integer
			Dim total As Integer = 0
			conexao = Rotinas.getConexao()
			cmd = New SQLiteCommand("select count(*) from Abas", conexao)
			dr = cmd.ExecuteReader()

			If dr.Read() Then
				total = dr.GetInt32(0)
			End If
			Me.fecharSqlPreparado()

			Return total
		End Function

		Public Function consultarTudo(progressoLog As IProgressoLog) As List(Of Aba)
			Dim obj As Aba
			Dim lista As New List(Of Aba)()
			Dim pb As Progresso
			Dim i As Integer = 0
			Dim total As Integer = Me.consultarTotal()

			conexao = Rotinas.getConexao()
			cmd = New SQLiteCommand("select cod, nome from Abas", conexao)
			dr = cmd.ExecuteReader()

			If total > 0 Then
				pb = New Progresso()
				pb.Minimo = 0
				pb.Maximo = total - 1
				pb.Posicao = 0
				pb.Passo = 1

				While dr.Read()
					obj = novoObjeto(dr)
					obj.Ordem = i + 1

					lista.Add(obj)

					pb.Posicao = i

					If progressoLog IsNot Nothing Then
						progressoLog.ProgressoLog(pb)
					End If

					i += 1
				End While
			End If

			Me.fecharSqlPreparado()

			Return lista
		End Function

		Public Function incluir(obj As Aba) As Integer
			Dim ret As Integer
			conexao = Rotinas.getConexao()
			cmd = New SQLiteCommand("insert into Abas(cod, nome) values(@1,@2)", conexao)
			cmd.Parameters.AddWithValue("@1", obj.Codigo)
			cmd.Parameters.AddWithValue("@2", obj.Nome)
			cmd.Prepare()
			ret = cmd.ExecuteNonQuery()
			Return ret
		End Function

		Public Function alterar(obj As Aba) As Integer
			Dim ret As Integer
			conexao = Rotinas.getConexao()
			cmd = New SQLiteCommand("update Abas set nome=@1 where cod=@2", conexao)
			cmd.Parameters.AddWithValue("@1", obj.Nome)
			cmd.Parameters.AddWithValue("@2", obj.Codigo)
			cmd.Prepare()
			ret = cmd.ExecuteNonQuery()
			Return ret
		End Function

		Public Function excluir(codigo As Integer) As Integer
			Dim ret As Integer
			conexao = Rotinas.getConexao()

			cmd = New SQLiteCommand("delete from Diretorios where aba=@1", conexao)
			cmd.Parameters.AddWithValue("@1", codigo)
			cmd.Prepare()
			cmd.ExecuteNonQuery()

			cmd = New SQLiteCommand("delete from Abas where cod=@1", conexao)
			cmd.Parameters.AddWithValue("@1", codigo)
			cmd.Prepare()
			ret = cmd.ExecuteNonQuery()

			Return ret
		End Function

		Public Function criarTabela() As Integer
			Dim ret As Integer
			conexao = Rotinas.getConexao()
			cmd = New SQLiteCommand("create table IF NOT EXISTS Abas(" & "cod integer not null, nome varchar(10) not null, " & "primary key (cod))", conexao)
			ret = cmd.ExecuteNonQuery()
			Return ret
		End Function

	End Class
End Namespace
